src/validator.ts

```diff
@@ -1,22 +1,61 @@
 import {
   FieldValidationFunctionSync,
   parseMessageWithCustomArgs,
+  ValidationResult,
 } from '@lemoncode/fonk';
 
-// TODO: Add validator type
-const VALIDATOR_TYPE = '';
+export const VALIDATOR_TYPE = 'BIC_VALIDATOR';
 
-// TODO: Add default message
-let defaultMessage = '';
+let defaultMessage = 'This is not in BIC correct format';
 export const setErrorMessage = message => (defaultMessage = message);
 
-const isDefined = value => value !== void 0 && value !== null && value !== '';
+const isDefined = (value: string) =>
+  value !== void 0 && value !== null && value !== '';
 
-export const validator: FieldValidationFunctionSync = fieldValidatorArgs => {
+export const validator: FieldValidationFunctionSync = (
+  fieldValidatorArgs
+): ValidationResult => {
   const { value, message = defaultMessage, customArgs } = fieldValidatorArgs;
 
-  // TODO: Add validator
-  const succeeded = !isDefined(value) || ...;
+  const matchAlfhanumericExpresion = (value: string): boolean => {
+    const letterAndNumbers = /^[0-9a-zA-Z]+$/;
+    return value.match(letterAndNumbers) ? true : false;
+  };
+  const hasCorrectLength = (value: string): boolean => {
+    return value.length === 8 || value.length === 11;
+  };
+  const validFormatBusinessPartyIdentifier = (
+    value: string,
+    prefix: boolean
+  ): boolean => {
+    if (prefix) {
+      value = value.substring(0, 4);
+    } else {
+      value = value.substring(6, 8);
+    }
+    return matchAlfhanumericExpresion(value);
+  };
```

Comment:
I think you can just do a first prefilter using a regular expression, this is just and idea

https://en.wikipedia.org/wiki/ISO_9362

\w{4}\w{2}\.{2}\.{3}?

4 letters bank
2 letters country
2 letters (maybe numeric or not)
3 letters or digits or XXX

You will need to fine tune that regEx but we will be able to remove a lot of code for this validator

Reply:
I have been testing some regular expressions but I can't find the perfect one! :(